test/unit/transport_test.rb

```diff
@@ -0,0 +1,94 @@
+# encoding: utf-8
+
+# rubocop: disable Metrics/BlockLength
+
+require_relative './helper'
+require 'train-winrm/transport'
+require 'logger'
+
+describe 'winrm transport' do
+  let(:cls) do
+    plat = Train::Platforms.name('mock').in_family('windows')
+    plat.add_platform_methods
+    Train::Platforms::Detect.stubs(:scan).returns(plat)
+    TrainPlugins::WinRM::Transport
+  end
+  let(:conf) {
+    {
+      host: rand.to_s,
+    password: rand.to_s,
+    logger: Logger.new(STDERR, level: :info),
+    }}
+  let(:cls_agent) { cls.new({ host: rand.to_s, logger: Logger.new(STDERR, level: :info) }) }
+
+  describe 'default options' do
+    let(:winrm) { cls.new({ host: 'dummy', logger: Logger.new(STDERR, level: :info) }) }
+
+    it 'can be instantiated (with valid config)' do
+      winrm.wont_be_nil
+    end
+
+    it 'configures the host' do
+      winrm.options[:host].must_equal 'dummy'
+    end
```

Comment:
Depending on the cost of the `winrm` let, it might be better to group all of these into one test.

Reply:
Should have added: If you run the this file with `-v` you'll get individual test times.